cmd/kubeops/internal/handler/handler_test.go

```diff
@@ -0,0 +1,135 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/kubeapps/kubeapps/pkg/agent"
+	"github.com/kubeapps/kubeapps/pkg/auth"
+	authFake "github.com/kubeapps/kubeapps/pkg/auth/fake"
+	chartFake "github.com/kubeapps/kubeapps/pkg/chart/fake"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chartutil"
+	kubefake "helm.sh/helm/v3/pkg/kube/fake"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+
+	"helm.sh/helm/v3/pkg/release"
+)
+
+const defaultListLimit = 256
+
+// newConfigFixture returns an agent.Config with fake clients
+// and memory storage.
+func newConfigFixture(t *testing.T) *agent.Config {
+	t.Helper()
+
+	return &agent.Config{
+		ActionConfig: &action.Configuration{
+			Releases:     storage.Init(driver.NewMemory()),
+			KubeClient:   &kubefake.FailingKubeClient{PrintingKubeClient: kubefake.PrintingKubeClient{Out: ioutil.Discard}},
+			Capabilities: chartutil.DefaultCapabilities,
+			Log: func(format string, v ...interface{}) {
+				t.Helper()
+				t.Logf(format, v...)
+			},
+		},
+		ChartClient: &chartFake.FakeChart{},
+		AgentOptions: agent.Options{
+			ListLimit: defaultListLimit,
+		},
+	}
+}
+
+func TestActions(t *testing.T) {
+	type testScenario struct {
+		// Scenario params
+		Description      string
+		ExistingReleases []release.Release
+		DisableAuth      bool
+		// Request params
+		RequestBody  string
+		RequestQuery string
+		Action       string
+		Params       map[string]string
+		// Expected result
+		StatusCode        int
+		RemainingReleases []release.Release
+		ResponseBody      string //optional
+	}
+
+	tests := []testScenario{}
+
+	for _, test := range tests {
+		// Initialize environment for test
+		req := httptest.NewRequest("GET", fmt.Sprintf("http://foo.bar%s", test.RequestQuery), strings.NewReader(test.RequestBody))
+		if !test.DisableAuth {
+			fauth := &authFake.FakeAuth{}
+			ctx := context.WithValue(req.Context(), auth.UserKey, fauth)
+			req = req.WithContext(ctx)
+		}
+		response := httptest.NewRecorder()
+		cfg := newConfigFixture(t)
+		for i := range test.ExistingReleases {
+			err := cfg.ActionConfig.Releases.Create(&test.ExistingReleases[i])
+			if err != nil {
+				t.Errorf("Failed to initiate test: %v", err)
+			}
+		}
+		// Perform request
+		t.Log(test.Description)
```

Comment:
Hmm, I'm not sure that the old tests in the `tiller-proxy/internal/handler/handler_test.go` are a good model to follow... as I see it does the same thing. Rather than `Log`ing the description of the test, the test should be run as a sub-test:
```
t.Run(test.Description, func(t *testing.T){
...
})
```

Take a look at the more recent tests that were added to the agent pkg at: https://github.com/kubeapps/kubeapps/blob/master/pkg/agent/agent_test.go#L185